### PR TITLE
add react nextjs example

### PR DESCRIPTION
**Pull request by dylanjha**
- [x] add examples/react-nextjs (`media-chrome` dependency points to locally built package)
- [x] add examples/react-nextjs to .npmignore so that it doesn't get published during `npm publish`

The main purpose of this is to give us an easy dev env for smoke testing and verify that future changes don't break React builds.

TODO:

- [ ] document how developers of media-chrome can use this example app
- [ ] make the example fancier (over-riding icons, using styled components, using styledjsx, etc.)

<details>
<summary> ✅ confirmed with `npm pack` that `.npmignore` is doing what we want:</summary>

❯ npm pack
npm notice
npm notice 📦  media-chrome@0.4.0
npm notice === Tarball Contents ===
npm notice 386B    examples/react-nextjs/.gitignore
npm notice 1.1kB   LICENSE
npm notice 275B    examples/react-nextjs/styles/globals.css
npm notice 1.7kB   examples/react-nextjs/styles/Home.module.css
npm notice 232B    examples/index.css
npm notice 1.6kB   examples/advanced.html
npm notice 1.5kB   examples/basic.html
npm notice 863B    examples/media-elements/dash.html
npm notice 230.9kB examples/themes/demuxed-2021-theme.html
npm notice 1.0kB   examples/media-elements/hls.html
npm notice 2.6kB   examples/index.html
npm notice 854B    examples/control-elements/media-captions-button.html
npm notice 1.6kB   examples/control-elements/media-clip-selector.html
npm notice 1.8kB   examples/control-elements/media-container.html
npm notice 827B    examples/control-elements/media-controller.html
npm notice 699B    examples/control-elements/media-current-time-display.html
npm notice 635B    examples/control-elements/media-duration-display.html
npm notice 712B    examples/control-elements/media-fullscreen-button.html
npm notice 1.9kB   examples/control-elements/media-mute-button.html
npm notice 598B    examples/control-elements/media-pip-button.html
npm notice 682B    examples/control-elements/media-play-button.html
npm notice 424B    examples/control-elements/media-seek-backward-button.html
npm notice 420B    examples/control-elements/media-seek-forward-button.html
npm notice 1.1kB   examples/control-elements/media-time-display.html
npm notice 645B    examples/control-elements/media-time-range.html
npm notice 634B    examples/control-elements/media-volume-range.html
npm notice 2.8kB   examples/mobile.html
npm notice 1.5kB   examples/media-elements/mux-video.html
npm notice 4.8kB   examples/themes/netflix-theme.html
npm notice 9.1kB   examples/slots-demo.html
npm notice 3.4kB   examples/themes/spotify-theme.html
npm notice 2.6kB   examples/standalone-controls.html
npm notice 4.0kB   examples/themes/youtube-theme.html
npm notice 862B    examples/media-elements/youtube.html
npm notice 25.9kB  examples/react-nextjs/public/favicon.ico
npm notice 137B    examples/react-nextjs/pages/_app.js
npm notice 1.1kB   dist/utils/captions.js
npm notice 1.9kB   dist/constants.js
npm notice 160B    dist/utils/defineCustomElement.js
npm notice 518B    dist/utils/fullscreenApi.js
npm notice 170B    examples/react-nextjs/pages/api/hello.js
npm notice 8.0kB   dist/extras/media-clip-selector/index.js
npm notice 1.9kB   dist/index.js
npm notice 5.4kB   dist/react/index.js
npm notice 2.3kB   examples/react-nextjs/pages/index.js
npm notice 690B    dist/labels/labels.js
npm notice 7.0kB   dist/media-captions-button.js
npm notice 3.1kB   dist/media-chrome-button.js
npm notice 5.3kB   dist/media-chrome-range.js
npm notice 5.3kB   dist/media-container.js
npm notice 1.5kB   dist/media-control-bar.js
npm notice 9.8kB   dist/media-controller.js
npm notice 492B    dist/media-current-time-display.js
npm notice 480B    dist/media-duration-display.js
npm notice 2.0kB   dist/media-fullscreen-button.js
npm notice 3.1kB   dist/media-mute-button.js
npm notice 1.7kB   dist/media-pip-button.js
npm notice 1.7kB   dist/media-play-button.js
npm notice 1.5kB   dist/media-playback-rate-button.js
npm notice 292B    dist/media-progress-range.js
npm notice 2.1kB   dist/media-seek-backward-button.js
npm notice 2.1kB   dist/media-seek-forward-button.js
npm notice 1.3kB   dist/media-settings-popup.js
npm notice 1.8kB   dist/media-text-display.js
npm notice 4.6kB   dist/themes/media-theme-netflix.js
npm notice 384B    dist/themes/media-theme.js
npm notice 1.9kB   dist/media-thumbnail-preview.js
npm notice 1.5kB   dist/media-time-display.js
npm notice 3.7kB   dist/media-time-range.js
npm notice 428B    dist/media-title-element.js
npm notice 1.2kB   dist/media-volume-range.js
npm notice 46B     examples/react-nextjs/next.config.js
npm notice 434B    dist/utils/server-safe-globals.js
npm notice 318B    dist/utils/stringUtils.js
npm notice 793B    dist/utils/time.js
npm notice 1.6kB   dist/react/common/utils.js
npm notice 40B     examples/react-nextjs/.eslintrc.json
npm notice 388B    examples/react-nextjs/package.json
npm notice 1.3kB   package.json
npm notice 73.1kB  examples/react-nextjs/yarn.lock
npm notice 1.6kB   examples/react-nextjs/README.md
npm notice 12.3kB  README.md
npm notice 1.1kB   examples/react-nextjs/public/vercel.svg
npm notice 28.7kB  examples/.basic.html.swp
npm notice 24.6kB  examples/react-nextjs/pages/.index.js.swp
npm notice 12.3kB  examples/react-nextjs/.package.json.swp
npm notice 720B    examples/vtt/en-cc.vtt
npm notice 632B    examples/vtt/en-sub.vtt
npm notice 612B    examples/vtt/es-sub.vtt
npm notice === Tarball Details ===
npm notice name:          media-chrome
npm notice version:       0.4.0
npm notice filename:      media-chrome-0.4.0.tgz
npm notice package size:  138.3 kB
npm notice unpacked size: 557.6 kB
npm notice shasum:        cb538fdb25c736601ef736432e85a89a5a5eea60
npm notice integrity:     sha512-HTW/wbQPxfzkI[...]LutBTkMbqENTw==
npm notice total files:   89
npm notice
media-chrome-0.4.0.tgz

</details>

### CODE REVIEW

File: examples/react-nextjs/package.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
+  },
+  "dependencies": {
+    "media-chrome": "file:../../",
+    "next": "12.0.10",
```
Comment (cjpillsbury) [inline examples/react-nextjs/package.json:12]: This should probably be a TypeScript app to get as much "bang for our buck" with smoke testing.

**Review by cjpillsbury — CHANGES_REQUESTED** [review]
A couple of callouts. If it would be easier, I can use this for reference and just migrate my test app over. Overall, 🙏 🎉

File: examples/react-nextjs/package.json (hunk 1 above)
Comment (dylanjha) [reply examples/react-nextjs/package.json:12]: Ahh, I was on the fence about that.  You're right though. I'll update to TS

File: yarn.lock [hunk 2]
```diff
@@ -3,63 +3,63 @@
 
 
 "@babel/code-frame@^7.0.0", "@babel/code-frame@^7.12.11":
-  "integrity" "sha512-9pzDqyc6OLDaqe+zbACgFkb6fKMNG6CObKpnYXChRsvYGyEdc7CA2BaqeOM+vOtCS5ndmJicPJhKAwYRI6UfFw=="
```
[the remark this answers is not in the document]
Comment (dylanjha) [reply yarn.lock]: woops

**Comment by cjpillsbury** [comment]
Used this as a reference for adding in a separate PR.

**Pull request closed without merging**